### PR TITLE
feat(i18n, client, curriculum): enable traditional chinese

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #XXXXX

This pull request will be implementing the automatic translation tool to convert the Simplified Chinese files into Traditional Chinese files.

- [X] Implement Script - [Example PR](https://github.com/nhcarrigan/actions-testing/pull/30) - [Example Action Run](https://github.com/nhcarrigan/actions-testing/actions/runs/658713671)
  - [X] Merge action into existing workflows, per @RandellDawson's [suggestion](https://github.com/nhcarrigan/actions-testing/pull/27#issuecomment-799920142)?
  - [ ] Auto-request review from freecodecamp/Chinese team to validate parser is working as expected?
- [X] Set up config
- [ ] Prepare deployment pipelines with @raisedadead 
  - [ ] Should deploy on root `freecodecamp.org` domain with `/chinese-traditional` path? ~~`/chinese-classical` path, [mirroring Wikipedia](https://zh-classical.wikipedia.org/wiki/%E7%B6%AD%E5%9F%BA%E5%A4%A7%E5%85%B8:%E5%8D%B7%E9%A6%96) to an extent (discussed this with @QuincyLarson).~~
- [ ] Upload initial files? Or allow action to generate them?

### CODE REVIEW

**Comment by RandellDawson** [comment]
Just a clarification for me.  Will there only be one Chinese language for the site?

**Comment by nhcarrigan** [comment]
There will be two - an offering of our hand-translated pages in Simplified Chinese, and a machine-converted Traditional Chinese view.

**Comment by RandellDawson** [comment]
Then why not call it `chinese-traditional` vs. `chinese-classical`?

**Comment by nhcarrigan** [comment]
Because when Quincy and I were discussing where to deploy this, we looked at how Wikipedia does it.

I **thought** they hosted on `zh-classical.wikipedia.org` and that's what I told Quincy, but taking another look that might be "Classical Chinese" which is potentially something different?

I'll switch it back to `chinese-traditional` as I do think that makes more sense.

**Comment by nhcarrigan** [comment]
(I'm failing a CI either way 😅 )

File: config/i18n/all-langs.js [hunk 1]
```diff
@@ -32,7 +34,8 @@ const langDisplayNames = {
 const langCodes = {
   english: 'en-US',
   espanol: 'es-419',
-  chinese: 'zh'
+  chinese: 'zh',
```
Comment (RandellDawson) [inline config/i18n/all-langs.js:37]: I am wondering if this should actually be:

```suggestion
  chinese: 'zh'-CN,
```

Comment (nhcarrigan) [reply config/i18n/all-langs.js:37]: According to https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry (which comes from the MDN docs), `zh-CN` isn't listed as a valid option. Simplified Chinese would be `zh-Hans` it seems.

**Comment by RandellDawson** [comment]
Changing `client/i18n/locales/chinese-traditional/intro.json` is causing the one GitHub action to fail.

**Comment by nhcarrigan** [comment]
The problem is if I don't upload that file, then the linter fails because the file doesn't exist. It's a one-off situation, really, since we wouldn't normally have a language we add manually.

**Review by raisedadead — DISMISSED** [review]

**Comment by raisedadead** [comment]
Landed via e46e80e08ffad8f0eb6dee4cf7fd43fdce4d6ba6

**Pull request closed without merging**